### PR TITLE
Long term match fragment + Time Availability Algorithm

### PR DESCRIPTION
**Pull request by carolyn-wang**
* Inflating design for Long Term Match fragment
* Calculates longest periods of overlap between 2 user availabilities and displays top 3 ideal times to meet up.
* Availability algorithm:
     * Compares 2 Boolean ArrayLists with user availabilities (`findConsecutiveRanges()`)
     * Sorts ranges by longest to shortest length (`findConsecutiveRanges()` line 223)
     * Converts top 3 longest ranges of overlapping availability into displayable Strings (`getMatchTimes()`)
    
Screen Shot 2022-07-22 at 4 53 38 PM

Future Updates:
* Algorithm to select ideal hangout location (current setting location randomly)
* Inflate coupon text

### CODE REVIEW

File: app/src/main/java/com/example/friendly/MatchingUtils.java [hunk 1]
```diff
@@ -85,6 +94,14 @@ public static Map<Double, ParseUser> getSortedMatches() {
...
         }
 
+        ArrayList<int[]> arr = findConsecutiveRanges(
+                Arrays.asList(true, true, false, true, true, true, false, false, false, false, false, false, true, true, true, true, false),
+                Arrays.asList(true, true, true, true, true, false, false, false, false, false, false, true, true, true, true, false, false));
+
+        for (int[] ints : arr) {
+            Log.i(TAG, ints[0] + " " + ints[1]);
+        }
+
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/MatchingUtils.java:104]: This looks like debug code that should be removed

**Review by NavinMahabir — COMMENTED** [review]
Refraining from approving because unsure if there is still debug code in the PR

File: app/src/main/java/com/example/friendly/utils/MatchingUtils.java [hunk 2]
```diff
@@ -164,16 +168,12 @@ private static double getIntSimilarityScore(ParseUser nearbyUser, String listKey
...
-    }
-
-
     /**
-     * Finds intersection between user availabilities and returns number of overlapping hours
+     * Finds indicies ranges where both array values are true
+     * and returns sorted ArrayList<int[]> with longer ranges towards the beginning of the list.
+     * rangeIndex tracks the index for adding ranges to availabilityRanges.
      *
+<<<<<<< HEAD
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/utils/MatchingUtils.java:176]: looks like you have an artifact from your rebase

Comment (carolyn-wang) [reply app/src/main/java/com/example/friendly/utils/MatchingUtils.java]: Just fixed

**Review by NavinMahabir — APPROVED** [review]
Looks good :)

**Pull request merged**